client/performance-profiler/components/core-web-vitals-display/index.tsx

```diff
@@ -61,8 +61,16 @@ export const CoreWebVitalsDisplay = ( props: CoreWebVitalsDisplayProps ) => {
 
 	const dataAvailable = metrics.length > 0 && metrics.some( ( item ) => item !== null );
 	const historicalData = metrics.map( ( item, index ) => {
+		let formattedDate: unknown;
+		if ( typeof dates[ index ] === 'string' ) {
+			formattedDate = dates[ index ]; // this is to ensure compability with reports before https://code.a8c.com/D159137
+		} else {
```

Comment:
Considering everything we have until now should be test data wouldn't be easier to delete all the previous reports and be able to remove this condition?

Reply:
Please remember the following text:
There are two paths from here:

#### Start Fresh:
We will delete all records in the database and start again. What is the impact of this? Some stakeholders might not be able to view their saved reports. We can quickly mitigate this by giving them a heads-up. The benefit is that we can get rid of this condition.

#### Keep Backward Compatibility:
In this path, we delete records for the old site profiler but keep the newer ones. This has no impact on stakeholders, and we can't get rid of this condition.

Based on the above, I would choose the first path.

Reply:
Thanks for detailing the two paths! Seeing the ramifications of both, I agree it's better to proceed with the first one. Let's keep this change in mind though the next time we're thinking about cleaning the DB of past runs. 🙂